mongodbatlas/atlas_region_helper.go

```diff
@@ -0,0 +1,303 @@
+package mongodbatlas
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type AtlasRegion string
+
+const (
+	// AWS:
+	// Americas
+	UsEast1    AtlasRegion = "US_EAST_1"
+	UsWest2    AtlasRegion = "US_WEST_2"
+	CaCentral1 AtlasRegion = "CA_CENTRAL_1"
```

Comment:
I really don't like the idea of maintaining this list 😿  One major drawback I see with having a hardcoded list is that whenever Atlas introduces a new region, users won't be able to utilize it until we manually update the region list in the Terraform code.


Could we investigate if we can update [listCloudProviderRegions](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Clusters/operation/listCloudProviderRegions) (or create a new endpoint in Atlas) to return the information that we need? The idea is that we already have an enum with all the regions in the MMS codebase so that when the Atlas team adds a new region, we can have this info in TF for free which will prevent us from manually maintaining this list. 

Once we have the endpoint, the TF cluster resource will just need to make an additional API call to get the correct region name.

I understand this means dedicating more time to this ticket. Tagging @colm-quinn to see if he agrees on investigating this other solution.

Reply:
While I don't see the cost of maintaining this mapping to be a lot as we don't expect new regions to be added very often, I'm okay with adding a new endpoint for this mapping so it can be available to a broader audience and move the maintenance to MMS if it already has region enums.

For my knowledge, how would the ownership of that API be worked out? (since you mentioned about an Atlas team)

Reply:
@andreaangiolillo Came across [this comment](https://jira.mongodb.org/browse/CLOUDP-123180?focusedCommentId=4841378&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4841378) in a similar JIRA ticket for Atlas CLI. Do you remember more about this? Can this existing endpoint already return all region mappings?

Reply:
The ticket is referring to this endpoint https://github.com/10gen/mms/pull/38225, which is the same one I shared with you via Slack. However, it seems that we didn't actually implement the desired functionality for that endpoint. Instead, we only created the related AtlasCLI command. If you need more information about the ticket, feel free to contact Bianca

Reply:
Synced with @blva offline, as per her knowledge, the endpoint will return only list of regions by instance size and not mapping with Atlas regions.

Reply:
If we can leverage the enums/list defined in Atlas to do this via an extension to the existing API, then I think it's a good approach and example of leverage, where we solve once for all integrations.

When we originally spoke about the mapping conversion here, we were hoping to generate the names by transformation, rather than by maintaining a manual list. Def can see a maintaneance issue & source of bugs going forward with the current implementat

Reply:
To add, I can see the mapping being in the generated SDK have a lot of user value, again a reason to extend the API

Reply:
> When we originally spoke about the mapping conversion here, we were hoping to generate the names by transformation, rather than by maintaining a manual list. 

Aah I see, @colm-quinn generating by transformation was never an option [as noted here](https://jira.mongodb.org/browse/CLOUDP-123180?focusedCommentId=4841378&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4841378) 

I have started looking into the Atlas endpoint for this mapping.